modules/masterbar/admin-menu/class-admin-menu.php

```diff
@@ -0,0 +1,725 @@
+<?php
+/**
+ * Admin Menu file.
+ *
+ * @package A8C\Admin_Plugins
+ */
+
+namespace Automattic\Jetpack\Dashboard_Customizations;
+
+/**
+ * Class Admin_Menu.
+ */
+class Admin_Menu {
+	/**
+	 * Holds class instance.
+	 *
+	 * @var Admin_Menu
+	 */
+	private static $instance;
+
+	/**
+	 * Whether the current request is a REST API request.
+	 *
+	 * @var bool
+	 */
+	private $is_api_request;
+
+	/**
+	 * Admin_Menu constructor.
+	 */
+	private function __construct() {
+		if ( jetpack_is_atomic_site() ) {
+			add_action(
+				'admin_menu',
+				function () {
+					remove_action( 'admin_menu', 'gutenberg_menu', 9 );
+				},
+				0
+			);
+		}
+
+		add_action( 'admin_menu', array( $this, 'reregister_menu_items' ), 99999 );
+		add_action( 'admin_enqueue_scripts', array( $this, 'enqueue_scripts' ) );
+	}
+
+	/**
+	 * Returns class instance.
+	 *
+	 * @return Admin_Menu
+	 */
+	public static function get_instance() {
+		if ( null === static::$instance ) {
+			static::$instance = new self();
+		}
+
+		return static::$instance;
+	}
+
+	/**
+	 * Create the desired menu output.
+	 */
+	public function reregister_menu_items() {
+		$this->is_api_request = ( defined( 'REST_API_PLUGINS' ) && REST_API_PLUGINS ) || ( defined( 'REST_REQUEST' ) && REST_REQUEST );
+
+		$domain = wp_parse_url( get_home_url(), PHP_URL_HOST );
```

Comment:
I think this (here and downer) may prove problematic on sites that live in a subdirectory. Maybe we should use something like this?
https://github.com/Automattic/jetpack/blob/acfd44811fd55e3ead09648ab95adae92ce860b5/modules%2Fcalypsoify%2Fclass.jetpack-calypsoify.php#L380-L399

Reply:
It looks like there are currently five instances of that site suffix determination in Jetpack:
https://github.com/Automattic/jetpack/search?q=get_calypso_site_slug

With Calypsoify going away soon, where would be a good place to create a utility function for it (before I create a sixth instance in this class)?

Reply:
👍
`functions.global.php` may be a good place for it. It may actually be a good time to move `Jetpack`'s `build_raw_urls` there as well.